### PR TITLE
chore(ci): CI job to sync bigquery-etl submodule in telemetry-airflow-dags

### PR DESCRIPTION
**Pull request by mikaeld**
# Description

Adds a CircleCI job to synchronize/update the `bigquery-etl` submodule in [telemetry-airflow-dags](https://github.com/mozilla/telemetry-airflow-dags/). A SSH deploy key with read/write access was added to the CircleCI project.

Checklist for reviewer:

- [x] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [x] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [x] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [x] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [x] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1596)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -405,6 +405,28 @@ jobs:
...
+      - add_ssh_keys:
+          fingerprints:
+            - "e4:30:50:41:53:f0:d6:3a:bb:c9:38:54:2d:ca:56:41"
+      - run:
+          name: 🤖 Update DAGs repository
+          command: |
+            ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+            git config --global user.email "dataops@mozilla.com"
+            git config --global user.name "CircleCI Job"
+            git clone --branch main git@github.com:mozilla/telemetry-airflow-dags.git
```
Comment (scholtzan) [inline .circleci/config.yml:421]: Could we use `generated-sql` instead of `main` here?

**Review by scholtzan — APPROVED** [review]
lgtm

File: .circleci/config.yml (hunk 1 above)
Comment (mikaeld) [reply .circleci/config.yml:421]: `mozilla/telemetry-airflow-dags` uses submodules in the `main` branch. Any specific reason why we would use a different branch?

Comment (mikaeld) [reply .circleci/config.yml:421]: Actually, your comment made me realise we don't pull the right branch.

`git pull origin dags` is a copy-paste error from `telemetry-airflow` and should be `git pull origin generated-sql` instead.

Comment (mikaeld) [reply .circleci/config.yml:421]: Yes, we don't use DAGs from bigquery-etl's main branch anymore. The setup is the following
image

where CI from three repos update their respective git submodules in `telemetry-airflow-dags` main branch.

Comment (mikaeld) [reply .circleci/config.yml:421]: You can see the repo here https://github.com/mozilla/telemetry-airflow-dags which has been configured for telemetry-airflow and bigquery-etl. private-bigquery-etl is next

**Pull request merged**